### PR TITLE
Link waitfor doc

### LINKED ISSUE #51971: Update waitFor() documentation
link the API docs on `waitFor()` to the docs for `runUnsynchronized()` (https://api.flutter.dev/flutter/flutter_driver/FlutterDriver/runUnsynchronized.html), which has much better guidance.

Noted from guidance on https://buganizer.corp.google.com/issues/150285048 (internal bug).

cc @zanderso

### PR DESCRIPTION
**Pull request by darshankawar**
## Description

This PR adds a reference link of Flutter Driver's [runUnSynchronized()](https://api.flutter.dev/flutter/flutter_driver/FlutterDriver/runUnsynchronized.html) method in [waitFor()](https://api.flutter.dev/flutter/flutter_driver/FlutterDriver/waitFor.html) method doc, since the former method has better detailing.

## Related Issues

Fixes [51971](https://github.com/flutter/flutter/issues/51971)

## Tests

This does not need tests, as it is purely a documentation PR.

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by darshankawar** [comment]
@googlebot I signed CLA

**Comment by darshankawar** [comment]
@googlebot I fixed it.

**Comment by DK15** [comment]
@googlebot I consent.

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 1]
```diff
@@ -200,6 +200,11 @@ abstract class FlutterDriver {
     await sendCommand(WaitFor(finder, timeout: timeout));
   }
 
+  /// See also:
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:203]: The "see also" section needs to go to the end of the comment.

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 2]
```diff
@@ -200,6 +200,11 @@ abstract class FlutterDriver {
...
+  ///
+  /// * [FlutterDriver.runUnsynchronized], which executes tests by disabling
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:205]: It's not clear why runUnsynchronized is relevant here?

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 3]
```diff
@@ -675,6 +675,9 @@ abstract class FlutterDriver {
   /// With frame sync disabled, it's the responsibility of the test author to
   /// ensure that no action is performed while the app is undergoing a
   /// transition to avoid flakiness.
+  /// See also:
+  ///
+  /// * [FlutterDriver.waitFor], which waits until target is located.
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:680]: Based on this comment its unclear to me why I should look at waitFor and how it relates to runUnsynchronized,

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 4]
```diff
@@ -675,6 +675,9 @@ abstract class FlutterDriver {
   /// With frame sync disabled, it's the responsibility of the test author to
   /// ensure that no action is performed while the app is undergoing a
   /// transition to avoid flakiness.
+  /// See also:
```
[the remark this answers is not in the document]
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart:679]: @goderbauer added.

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 5]
```diff
@@ -199,7 +199,7 @@ abstract class FlutterDriver {
   Future<void> waitFor(SerializableFinder finder, { Duration timeout }) async {
     await sendCommand(WaitFor(finder, timeout: timeout));
   }
-
+
```
[the remark this answers is not in the document]
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart:202]: I think I don't see the extra space when I tried to revert this. There's a single line space.

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 3 above)
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart:681]: @goderbauer Added this per the original issue [description](https://github.com/flutter/flutter/issues/51971#issue-575853935).

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 5 above)
[the remark this answers is not in the document]
Comment (Piinks) [reply packages/flutter_driver/lib/src/driver/driver.dart:202]: It looks like this trailing space is causing the analyzer to fail.

[the remark this answers is not in the document]
Comment (darshankawar) [reply packages/flutter_driver/lib/src/driver/driver.dart]: @Piinks fixed.

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 6]
```diff
@@ -199,7 +199,6 @@ abstract class FlutterDriver {
   Future<void> waitFor(SerializableFinder finder, { Duration timeout }) async {
     await sendCommand(WaitFor(finder, timeout: timeout));
   }
-
```
Comment (goderbauer) [inline packages/flutter_driver/lib/src/driver/driver.dart:202]: Revert this?

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 3 above)
Comment (goderbauer) [reply packages/flutter_driver/lib/src/driver/driver.dart:680]: The issue is asking for the link in the opposite direction: `waitFor` should link to `runUnsynchronized` and needs to describe how the two concepts relate.

**Comment by goderbauer** [comment]
@darshankawar Do you still have plans to come back to this PR?

**Comment by darshankawar** [comment]
@goderbauer sorry for delay. Yes I am planning to update it per your feedback. Will update in coming days.

**Comment by goderbauer** [comment]
@darshankawar I am going to close this for now. Feel free to re-open when it is updated and ready for review again.

**Pull request closed without merging**